docs/learn/migrate/solana.mdx

```diff
@@ -1,8 +1,689 @@
+
+The Stellar CLI is your gateway to the Stellar smart contract ecosystem. Here are the installation options:
+
+```bash
+# Install Rust (if you haven't already)
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+
+# Install Stellar CLI - choose your preferred method:
+# Option 1: Homebrew (macOS, Linux)
+brew install stellar-cli
+
+# Option 2: Windows Package Manager
+winget install --id Stellar.StellarCLI
+
+# Option 3: From source using Cargo
+cargo install --locked stellar-cli
+
+# Verify the installation
+stellar --version
+```
+
+### Project Setup
+
+Creating a new project is straightforward:
+
+```bash
+# Create a new project
+cargo new my-soroban-contract
+cd my-soroban-contract
+
+# Add Soroban dependencies to Cargo.toml
+```
+
+## Migrating Your First Contract
+
+Let's walk through the migration process step by step. We'll start with a simple example and gradually build up to more complex patterns.
+
+### Step 1: Update Dependencies
+
+First, you'll need to replace your Solana dependencies with Stellar equivalents. Here's what your `Cargo.toml` should look like:
+
+```toml
+[package]
+name = "my-soroban-contract"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+soroban-sdk = "$VERSION"
+
+[dev_dependencies]
+soroban-sdk = { version = "$VERSION", features = ["testutils"] }
+```
+
+:::note **Important**:
+
+Replace `$VERSION` with the actual latest version from [crates.io](https://crates.io/crates/soroban-sdk). As of the latest Stellar documentation, the current stable version is v20.4.0, but you should always check for the most recent release.
+
+:::
+
+The release profile configuration is crucial for keeping your contract size under the 64KB limit:
+
+```toml
+[profile.release]
+opt-level = "z"
+overflow-checks = true
+debug = 0
+strip = "symbols"
+debug-assertions = false
+panic = "abort"
+codegen-units = 1
+lto = true
+
+[profile.release.build-override]
+opt-level = 3
+incremental = false
+codegen-units = 1
+```
+
+### Step 2: Convert Program Entry Point
+
+The biggest change you'll notice is how functions are defined and called.
+
+**Before (Solana):**
+
+```rust
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    entrypoint,
+    entrypoint::ProgramResult,
+    msg,
+    program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+entrypoint!(process);
+
+pub fn process(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+    let payer = next_account_info(accounts_iter)?;
+
+    msg!("Hello from Solana!");
+    Ok(())
+}
+```
+
+**After (Stellar):**
+
+```rust
+use soroban_sdk::{
+    contract, contractimpl, symbol_short, vec, Address, Env, Symbol, Vec,
+};
+
+#[contract]
+pub struct HelloContract;
+
+#[contractimpl]
+impl HelloContract {
+    pub fn hello(env: Env, to: Symbol) -> Vec<Symbol> {
+        vec![&env, symbol_short!("Hello"), to]
```

Comment:
I would suggest using `String` instead of `Symbol` everywhere (we do that in the updated hello_world example as well: https://github.com/stellar/soroban-examples/blob/main/hello_world/src/lib.rs)